cartographer_rviz/cartographer_rviz/drawable_submap.cc

```diff
@@ -25,11 +25,16 @@
 #include "Eigen/Geometry"
 #include "OgreGpuProgramParams.h"
 #include "OgreImage.h"
+#include "cartographer/common/make_unique.h"
 #include "cartographer/common/port.h"
 #include "cartographer_ros_msgs/SubmapQuery.h"
 #include "eigen_conversions/eigen_msg.h"
 #include "ros/ros.h"
 
+#if (QT_VERSION < QT_VERSION_CHECK(5, 0, 0))
```

Comment:
QStringLiteral has been introduced in Qt5 for compile-time string formatting. On indigo, we are building against Qt4. QStringLiteral is not defined there at all. I copied the fallback for compilers which do not support compile-time string formatting from Qt5's version of `qstring.h`.

We could just alias QStringLiteral to QString, but I thought it would be a better idea to [copy their fallback](http://stackoverflow.com/a/21412477/1290505) from `qstring.h`:

```
#ifndef QStringLiteral
// no lambdas, not GCC, or GCC in C++98 mode with 4-byte wchar_t
// fallback, return a temporary QString
// source code is assumed to be encoded in UTF-8

# define QStringLiteral(str) QString::fromUtf8("" str "", sizeof(str) - 1)
#endif
```